JOSESwift/Sources/CryptoImplementation/EC.swift

```diff
@@ -95,33 +95,46 @@ public enum ECCurveType: String, Codable {
     }
 }
 
-fileprivate extension SignatureAlgorithm {
-    typealias DigestFunction = (
-            UnsafeRawPointer?,
-            UInt32,
-            UnsafeMutablePointer<UInt8>?
-    ) -> UnsafeMutablePointer<UInt8>?
-
-    func createDigest(input: Data) throws -> [UInt8] {
-        guard
-                let computedDigestLength = digestLength,
-                let computedDigestFunction = digestFunction
-                else {
-            throw ECError.invalidCurveDigestAlgorithm
+// converting integers to and from DER encoded ASN.1 as described here:
+// https://docs.microsoft.com/en-us/windows/desktop/seccertenroll/about-integer
+fileprivate extension Data {
+    func withLengthFixedTo(_ octetLength: Int) -> Data {
+        let varLength = self.count
+        if varLength > octetLength + 1 {
+            fatalError("Unable to parse ASN.1 Integer")
+        }
+        if varLength == octetLength + 1 {
+            return self.dropFirst()
         }
-        var digest = [UInt8](repeating: 0, count: computedDigestLength)
-        _ = computedDigestFunction(Array(input), UInt32(input.count), &digest)
-        return digest
+        if varLength == octetLength {
+            return self
+        }
+        if varLength < octetLength {
+            return Data.init(count: octetLength - varLength) + self
+        }
+        fatalError("Unable to parse ASN.1 Integer")
     }
 
-    var digestLength: Int? {
+    func fixedLengthToAsn1() -> Data {
+        assert(self.count > 0)
```

Comment:
Yes, asserts are omitted in release builds.

Instead of changing the assert, I will add an additional check directly in the verify method, so incorrect user code gets an error and the assert is only an internal thing.